_posts/2022-06-10-slick-intro.md

```diff
@@ -0,0 +1,465 @@
+```
+
+### 4.2. Querying Rows from Database
+Now, let's see how to execute select queries and fetch the rows from the table. To get all the movies, we can use the `movieTable` as:
+```scala
+val movies: Future[Seq[Movie]] = db.run(SlickTables.movieTable.result)
+```
+The method `.result` on the _movieTable_ will create an executable action. We can filter the rows in the table using the _filter_ method just like on a collection. However, we need to use `===` method instead. This method will compare the value provided with that in the database column. For this method to be available, we need to import _api_ from the profile.
+```
+import profile.api._
+```
+where profile is _PostgresProfile_ we used before.
+
+```scala
+def findMovieByName(name: String): Future[Option[Movie]] = {
+    db.run(SlickTables.movieTable.filter(_.name === name).result.headOption)
+}
+```
+
+### 4.3. Update a Row
+We can use the method `update` to modify a record after applying the filter. 
+```scala
+def updateMovie(movieId: Long, movie: Movie): Future[Int] = {
+    val updateQuery = SlickTables.movieTable.filter(_.id === movieId).update(movie)
+    db.run(updateQuery)
+}
+```
+In the above sample, if the filter by id is not applied, it will update all the records with the same value.
+
+If we want to update only a particular field, we can use the `map` to get the field and then update it. For example, to update the movie name, we can do as:
+```scala
+val updateMovieNameQuery = SlickTables.movieTable.filter(_.id === movieId).map(_.name).update("newName")
+``` 
+
+### 4.4. Delete a Row
+We can also delete a record from the database by using the method `delete`. To delete a movie we can do:
+```scala
+val deleteQuery = SlickTables.movieTable.filter(_.id === movieId).delete
+```
+
+## 5. Advanced Options
+Since we are become familiar with the basic CRUD operations, let's look at some more advanced concepts.
+
+### 5.1 Executing Plain Query
+Sometimes we might need to run plain SQL queries to get the results in a better way. Slick provides multiple ways to run the plain query. Let's look at a simple way to run the plain query. However, since we are using the plain query, we need to provide some additional information to slick to make the queries typesafe. For that, we need to provide an implicit value with the mappings.
+
+```scala
+def getAllMoviesByPlainQuery: Future[Seq[Movie]] = {
+  implicit val getResultMovie =
+    GetResult(r => Movie(r.<<, r.<<, LocalDate.parse(r.nextString()), r.<<))
+  val moviesQuery = sql"""SELECT * FROM public."Movie" """.as[Movie]
+  db.run(moviesQuery)
+}
+```
+
+The implicit _GetResult_ informs Slick on how to map the results of a plain query to required case class. _GetResult_ takes a lambda, which has the _ResultSet_ from the query. We need to provide the datatypes of the result fields. If we use `<<`, Slick will try to infer the type based on the mapped case class fields. 
```

Comment:
The `<<` thing is still unclear to me while reading this. I know it's quite tricky to explain, but we need to give the reader enough context. You can add something like "You can think of `<<` as a placeholder, which contains the correct type, given that it's placed in the relevant field of the Movie constructor. By using `<<`, Slick will then contain (at runtime) the right types to map to the table holding movie records."

Reply:
Hi Daniel, I made some changes. Can you check and see if this makes it clearer?